subsys/nrf_security/src/drivers/cracen/cracenpsa/src/key_derivation.c

```diff
@@ -193,7 +193,36 @@ psa_status_t cracen_key_derivation_setup(cracen_key_derivation_operation_t *oper
 {
 	operation->alg = alg;
 
-	if (IS_ENABLED(PSA_NEED_CRACEN_HKDF) && PSA_ALG_IS_HKDF(operation->alg)) {
+	if (IS_ENABLED(PSA_NEED_CRACEN_HKDF) && PSA_ALG_IS_HKDF(operation->alg) ||
+	PSA_ALG_IS_HKDF_EXPAND(operation->alg)) {
+		size_t hash_size = PSA_HASH_LENGTH(PSA_ALG_HKDF_GET_HASH(alg));
+
+		if (hash_size == 0) {
+			return PSA_ERROR_NOT_SUPPORTED;
+		}
```

Comment:
Is this needed?

Reply:
Oh yeah that can be removed.

Reply:
I meant the `if` and `return`, not the whole block. :slightly_smiling_face:

Reply:
I know, but I had a duplicate block there for PSA_ALG_IS_HKDF_EXPAND, which I had already folded into this one as there was no need for both. The 'if' is from the existing code, I think it is the first test to make sure the code has a valid hash alg that is called and psa_key_derivation_setup should return 'PSA_ERROR_NOT_SUPPORTED' when called on an unsupported algorithm, but I am unsure if 1. You can get to this point at all as 'PSA_ALG_HKDF(unsupported_alg)' return is undefined and 2. the hash alg being unsupported counts as the key_derivation alg is unsupported

Reply:
Yeah. I was thinking that this check might be covered by other stuff as well. But anyway it doesn't hurt, if we want to optimize ROM footprint at some point there seems to be some room for improvement in there anyway.

Reply:
Hmm not sure. I guess you want to make sure that `alg` is both a hash and supported.